src/data/mystery-encounter-option.ts

```diff
@@ -20,15 +27,94 @@ export default class MysteryEncounterOption implements MysteryEncounterOption {
   meetsRequirements?(scene: BattleScene) {
     return !this.requirements.some(requirement => !requirement.meetsRequirement(scene));
   }
+  meetsProtagonistRequirementAndProtagonistPokemonSelected?(scene: BattleScene) {
```

Comment:
Just re-iterating, can we make this function private and call it from meetsRequirements() instead?